ducktape/cluster/remoteaccount.py

```diff
@@ -82,40 +205,92 @@ def ssh_command(self, cmd):
         return r
 
     def ssh(self, cmd, allow_fail=False):
+        """Run the given command on the remote host, and block until the command has finished running.
```

Comment:
While we're rethinking APIs, I'd spend some time thinking about the right set of SSH primitives. As it stands today, we have a few different patterns. There's probably some baseline SSH method we can provide that provides good flexibility, then a small number of utilities built on top of that (and ideally better named than our current set). Originally when this started out I thought we'd be able to refactor easily, but we never got around to doing that before we split up the fmwk and